pkg/kopia/stream.go

```diff
@@ -146,6 +221,42 @@ func Read(ctx context.Context, backupID, path string, target io.Writer) error {
 	return errors.Wrap(err, "Failed to copy snapshot data to the target")
 }
 
+// ReadFile restores a kopia snapshot with the given ID to the given target
+func ReadFile(ctx context.Context, backupID, target string) error {
+	password, ok := repo.GetPersistedPassword(ctx, defaultConfigFilePath)
+	if !ok || password == "" {
+		return errors.New("Failed to retrieve kopia client passphrase")
+	}
+
+	rep, err := OpenRepository(ctx, defaultConfigFilePath, password, pullRepoPurpose)
+	if err != nil {
+		return errors.Wrap(err, "Failed to open kopia repository")
+	}
+
+	rootEntry, err := snapshotfs.FilesystemEntryFromIDWithPath(ctx, rep, backupID, false)
+	if err != nil {
+		return errors.Wrap(err, "Unable to get filesystem entry")
+	}
+
+	p, err := filepath.Abs(target)
+	if err != nil {
+		return errors.Wrap(err, "Unable to resolve path")
+	}
+	// TODO: Do we want to keep this flags configurable?
+	output := &restore.FilesystemOutput{
+		TargetPath:             p,
+		OverwriteDirectories:   true,
+		OverwriteFiles:         true,
+		OverwriteSymlinks:      true,
+		IgnorePermissionErrors: true,
```

Comment:
@pavannd1 do we want to have these flags configurable?

Reply:
Should be okay for now. We can improve later if the requirement comes up.